include/boost/static_string/static_string.hpp

```diff
@@ -5614,6 +5619,331 @@ operator+(
     std::size_t(0), +lhs);
 }
 
+
+//------------------------------------------------------------------------------
+//
+// std::string/std::string_view comparison
+//
+//------------------------------------------------------------------------------
+
+template<std::size_t N, typename CharT, typename Traits, typename Alloc>
+BOOST_STATIC_STRING_CPP14_CONSTEXPR
+inline
+bool
+operator==(
+  const std::basic_string<CharT, Traits, Alloc>& lhs,
+  const basic_static_string<N, CharT, Traits>& rhs)
+{
+  return detail::lexicographical_compare<CharT, Traits>(
+    lhs.data(), lhs.size(),
+    rhs.data(), rhs.size()) == 0;
+}
+
+template<std::size_t N, typename CharT, typename Traits, typename Alloc>
+BOOST_STATIC_STRING_CPP14_CONSTEXPR
+inline
+bool
+operator==(
+  const basic_static_string<N, CharT, Traits>& lhs,
+  const std::basic_string<CharT, Traits, Alloc>& rhs)
+{
+  return detail::lexicographical_compare<CharT, Traits>(
+    lhs.data(), lhs.size(),
+    rhs.data(), rhs.size()) == 0;
+}
+
+template<std::size_t N, typename CharT, typename Traits, typename Alloc>
+BOOST_STATIC_STRING_CPP14_CONSTEXPR
+inline
+bool
+operator!=(
+  const std::basic_string<CharT, Traits, Alloc>& lhs,
+  const basic_static_string<N, CharT, Traits>& rhs)
+{
+  return detail::lexicographical_compare<CharT, Traits>(
+    lhs.data(), lhs.size(),
+    rhs.data(), rhs.size()) != 0;
+}
+
+template<std::size_t N, typename CharT, typename Traits, typename Alloc>
+BOOST_STATIC_STRING_CPP14_CONSTEXPR
+inline
+bool
+operator!=(
+  const basic_static_string<N, CharT, Traits>& lhs,
+  const std::basic_string<CharT, Traits, Alloc>& rhs)
+{
+  return detail::lexicographical_compare<CharT, Traits>(
+    lhs.data(), lhs.size(),
+    rhs.data(), rhs.size()) != 0;
+}
+
+template<std::size_t N, typename CharT, typename Traits, typename Alloc>
+BOOST_STATIC_STRING_CPP14_CONSTEXPR
+inline
```

Comment:
function templates are automatically `inline`

Reply:
hmm, i was under impression that both template functions and inline functions are emitted as weak symbols, but template functions don't automatically get the inline hint (which the `inline` keyword provides).
i guess in most cases, the compiler will overrule the inline hint, anyway (though i think i've seen cases where the codegen was actually different)